infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationEventListener.java

```diff
@@ -103,10 +111,7 @@ private void processPendingRequests() {
         //  events handled by onNext(), this logic can be abstracted out to a common method which can be shared with
         //  onNext()
         for (CorfuStoreEntry event : pendingEvents) {
-            if (((ReplicationEvent)event.getPayload()).getType().equals(
```

Comment:
are we removing this check because other events get deleted once they are processed? 

If a regular forced snapshot sync(other than rolling upgrade) is received and fails, we should not re-trigger it IMO.  This is because the listener can get resubscribed much later as well.  So we require this check IMO.

Reply:
Will discuss this offline

Reply:
update: leaving this change like this for now. Have started an a thread offline to discuss the expectation here.

Reply:
triggerForcedSnapshotSyncForAllSessions() was written for the upgrade case but it is now being used in all forced snapshot sync scenarios.  To make it easy to understand and read, can you please cleanup the logic in onNext() where both events are processed in the same way?  Also, can you cleanup/modify the comment on L109?